### PR TITLE
Additem

### PR DESCRIPTION
**Pull request by neimanchen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "multer-s3": "^2.7.0",
     "nodemon-webpack-plugin": "^3.0.1",
     "pg": "^7.4.3",
+    "quagga": "^0.12.1",
```
Comment (skoshkarli) [inline package.json]: are we using this? if not, lets remove

Comment (neimanchen) [reply package.json]: its the barcode scanner library, still a WIP

File: src/client/components/AddItem/AddItem.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import UploadItem from './UploadItem.jsx';
+import Axios from 'axios';
+
+export class AddItem extends React.Component {
+  constructor(props) {
+    super(props);
+    //this.getBarcode = this.getBarcode.bind(this);
+  }
+
+  // componentDidMount() {
```
Comment (skoshkarli) [inline src/client/components/AddItem/AddItem.jsx]: can we moreve the commented out code?

Comment (skoshkarli) [reply src/client/components/AddItem/AddItem.jsx]: or is there a reason you are keeping it?

File: src/client/components/AddItem/AddItem.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+})
+
+const mapDispatchToProps = dispatch => ({
+  actions: bindActionCreators({ updateItem }, dispatch)
+});
+
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AddItem))
+
+
+  //   return (
```
Comment (skoshkarli) [inline src/client/components/AddItem/AddItem.jsx]: please cleandup unnecessary code

File: src/server/index.js [hunk 4]
```diff
@@ -62,6 +70,22 @@ function sendWeather(locationKey, res) {
...
 
+app.get('api/barcode', (req, res) => {
+  var config = {
+    headers: {
+      Authorization: barcodableKey
+    }
+  }
+  Axios.get('https://www.barcodable.com/api/v1/upc/826218178634', config)
+  .then((response) => {
+    console.log(response)
```
Comment (skoshkarli) [inline src/server/index.js]: remove the console log

File: src/server/index.js [hunk 5]
```diff
@@ -62,6 +70,22 @@ function sendWeather(locationKey, res) {
...
+    res.send(response);
+  }).catch((error) => {
+    res.status(500).send({error: 'There was an error getting your info from Barcodable'});
+    console.error(error);
```
Comment (skoshkarli) [inline src/server/index.js]: remove the console error

File: src/client/components/AddItem/AddItem.jsx (hunk 2 above)
Comment (neimanchen) [reply src/client/components/AddItem/AddItem.jsx]: things werent working and decided to scrap for MVP, will revisit

File: src/client/components/AddItem/UploadItemInfo.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import { withRouter } from 'react-router-dom';
+
+export class UploadItemInfo extends React.Component {
+  constructor(props) {
+    super(props)
+  }
+
+  render() {
+    return (
+      <Modal trigger={<Button>Show Modal</Button>} centered={false}>
```
Comment (hcnimi) [inline src/client/components/AddItem/UploadItemInfo.jsx]: I think we should rename to something more indicative of action.

File: src/client/index.jsx [hunk 7]
```diff
@@ -38,6 +39,7 @@ class App extends React.Component {
           <div className="content container">
             <Route path="/mycloset" component={MyCloset} store={store}/>
             <Route path="/closetboard" component={ClosetBoard} store={store}/>
+            <Route path="/additem" component={AddItem} store={store}/>
```
Comment (happysituations) [inline src/client/index.jsx]: Route here is now DefaultLayout as we discussed in zoom.

File: src/server/index.js [hunk 8]
```diff
@@ -62,6 +70,20 @@ function sendWeather(locationKey, res) {
     });
 }
 
+app.get('api/barcode', (req, res) => {
```
Comment (happysituations) [inline src/server/index.js]: api needs a "/" in the front

**Pull request merged**